src/lib/HST/Feature/Optimization.hs

```diff
@@ -17,45 +17,44 @@ import           HST.Environment.FreshVars      ( PM
 import           HST.Environment.Renaming       ( subst
                                                 , rename
                                                 )
-
-import qualified Language.Haskell.Exts.Syntax  as HSE
-import qualified Language.Haskell.Exts.Build   as B
+import qualified HST.Frontend.Syntax           as S
```

Comment:
I only used the `HST` alias to differentiate from the `HSE` syntax. I personally like the `S` alias more, not only because it is shorter, but also because it stands for syntax, so it is a very short abbreviation for the name of the module. `HST` on the other hand could refer to anything inside of `haskell-src-transformations` and I don't think it's clear that the syntax module is meant. In the Free Compiler, `IR` works an alias for the IR syntax, because the IR is a language, but that doesn't apply to `haskell-src-transformations`, where the new syntax module is only a part of HST.

Reply:
> I personally like the `S` alias more, not only because it is shorter, but also because it stands for syntax, so it is a very short abbreviation for the name of the module

Even though I'm not a fan of one letter aliases, it would be okay with me if you used `S` as an alias for `HST.Frontend.Syntax` long as you never import it as `HST` and you don't import other syntax modules as `S`.

> `HST` on the other hand could refer to anything inside of `haskell-src-transformations`

That's true, but I don't think that it is a problem. For example, I [use `HSE` as the alias](https://github.com/FreeProving/free-compiler/blob/477f0f5d898c34bc34827827895118a824127968/.hlint.yaml#L42) for anything in `haskell-src-exts`.

> `IR` works an alias for the IR syntax, because the IR is a language, but that doesn't apply to `haskell-src-transformations`

The intermediate representation of the Free Compiler also just started as an alternative Haskell AST. Through the addition of a parser, pretty printer and additional AST nodes, it slowly evolved into something that no one can deny is its own language. In case of `haskell-src-transformations` a parser and pretty printer will eventually become necessary as well if we want to write proper unit tests. If we recognize the AST as its own language, writing such a parser and pretty-printer gets much easier since we can ignore Haskell's layout rule and we can also add new language features if we ever need to. 

In fact, it wouldn't hurt if we gave the language a name. In the Free Compiler's it is a huge problem that IR was called Haskell initially. Some documentation as well as the module interfaces still refers to it IR as Haskell. Since our syntax is also just an intermediate representation, it would be fine in my opinion if we called it IR as well.